QuerySet.in_bulk returns Dict with values of correct model type.

syastrov opened the pull request:
- The keys are still Union[int, str] which doesn't cover all
possibilities, but it's not possible to cover without handling this in
the plugin, because the type of the Dict keys are dynamic depending on
which value for field_name you pass in.

syastrov committed:
QuerySet.in_bulk returns Dict with values of correct model type.

- The keys are still Union[int, str] which doesn't cover all
possibilities, but it's not possible to cover without handling this in
the plugin, because the type of the Dict keys are dynamic depending on
which value for field_name you pass in.

mkurnikov commented on django-stubs/db/models/query.pyi:
```diff
@@ -86,7 +86,7 @@ class QuerySet(Iterable[_T], Sized):
     def last(self) -> Optional[_T]: ...
     def in_bulk(
         self, id_list: Any = ..., *, field_name: str = ..., **kwargs: Any
```
Could you replace `id_list` parameter annotation with the `Sequence[Any]`, and `field_name` with `Optional[str]` while you're here?

syastrov replied:
Hi. I've changed the id_list annotation. I also made a few other improvements and added a test.

syastrov committed:
QuerySet.in_bulk fixes.

- Made id_list of type Sequence[Any], rather than Any, according to mkurnikov's review.
- Removed **kwargs.
- Made returned Dict keys of type Any rather than int or str as it depends on the provided field's type.
- Added test with examples from Django docs.

syastrov committed:
Merge branch 'master' into queryset_in_bulk

syastrov committed:
Change in_bulk id_list param to Iterable rather than Sequence.

syastrov committed:
Add possibility to pass list of test names as command-line arguments to typecheck_tests.py script.
